components/properties/src/runtime.rs

```diff
@@ -0,0 +1,436 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! This module contains utilities for working with properties where the specific property in use
+//! is not known at compile time.
+
+use crate::maps::CodePointMapData;
+use crate::names::PropertyValueNameToEnumMapper;
+use crate::sets::CodePointSetData;
+use crate::PropertiesError;
+
+/// This type can represent any Unicode property.
+///
+/// This is intended to be used in situations where the exact unicode property needed is
+/// only known at runtime, for example in regex engines.
+///
+/// The values are intended to be identical to ICU4C's UProperty enum
+#[allow(clippy::exhaustive_structs)] // newtype
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub struct UnicodeProperty(pub u32);
+
+#[allow(non_upper_case_globals)]
+impl UnicodeProperty {
```

Comment:
A choice I have made here is to include _every_ UProperty value here, not just ones ICU4X supports. The remaining values are largely useless since there are no lookup APIs for them, but I suspect such APIs will be useful to clients all the same.


We could comment them out if desired. I'm not sure if we should, leaning towards not doing so.

Reply:
also with the current choice the property kind matches don't have to deal with unknown properties, which is nice

Reply:
I think having the ability to match on these is still useful even if there isn't _ICU4X_ code for handling these